### PR TITLE
Upgrade LK/LGPU CI tests to use cuda 12.4

### PR DESCRIPTION
**Pull request by josephleekl**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [X] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
Lightning Kokkos CUDA C++ tests fail to compile due to [bug](https://forums.developer.nvidia.com/t/compiling-a-catch2-application-with-nvcc-std-c-20-leads-a-crash-in-cudafe/255088) with CUDA 12.2 + Catch2. CI instance is now upgraded to CUDA 12.4, and all GPU CI tests ([LK, LGPU/LTensor]x[C++, Python]) need to be updated to load CUDA 12.4.

**Description of the Change:**
Update github workflow to load CUDA 12.4. Also change Lightning Kokkos C++ test to use `Release` build instead of `RelWithDebInfo` due to segmentation fault when combined with coverage collection. The relevant test and line is [here](https://github.com/PennyLaneAI/pennylane-lightning/blob/e9a3b80823a2dc52f747ff20f91031512d0dbd63/pennylane_lightning/core/src/simulators/lightning_kokkos/gates/tests/Test_StateVectorKokkos_Param.cpp#L2062).

**Benefits:**
Will be able to run Lightning Kokkos C++ tests with CUDA.

**Possible Drawbacks:**

**Related GitHub Issues:**

[sc-80288]

### CODE REVIEW

**Review by AmintorDusko — APPROVED** [review]
This is great, thanks!

**Comment by AmintorDusko** [comment]
> This is great, thanks!

Before merging could you please run, stable/stable and latest/latest?

File: .github/workflows/tests_gpu_cpp.yml
Comment (maliasadi) [inline .github/workflows/tests_gpu_cpp.yml:1]: will you update [tests_lkcuda_cpp.yml](https://github.com/PennyLaneAI/pennylane-lightning/blob/master/.github/workflows/tests_lkcuda_cpp.yml) as part of this PR?

**Review by maliasadi — COMMENTED** [review]
LGTM! Thanks @josephleekl

File: .github/workflows/tests_gpu_cpp.yml
Comment (josephleekl) [reply .github/workflows/tests_gpu_cpp.yml:1]: I've accidentally *all* the changes when updating lkcuda_cpp when i put the paths back 🤦 I'll bring back the changes!

Comment (josephleekl) [reply .github/workflows/tests_gpu_cpp.yml:1]: I've brought back the file now

**Comment by josephleekl** [comment]
> > This is great, thanks!
> 
> Before merging could you please run, stable/stable and latest/latest?

Thanks Amintor! I'll update the latest/latest workflow to check this, as well as run the stable/stable

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -92,6 +92,9 @@
 
 ### Bug fixes
 
+* Update CUDA version to 12.4 for GPU tests on CI.
+  [(#1023)](https://github.com/PennyLaneAI/pennylane-lightning/pull/1018)
```
Comment (maliasadi) [inline .github/CHANGELOG.md:96]:
```suggestion
  [(#1023)](https://github.com/PennyLaneAI/pennylane-lightning/pull/1023)
```

File: .github/workflows/tests_lkcuda_cpp.yml [hunk 2]
```diff
@@ -190,7 +194,7 @@ jobs:
         run: |
           cd main
           cmake . -BBuild -G Ninja \
-            -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+            -DCMAKE_BUILD_TYPE=Release \
```
Comment (maliasadi) [inline .github/workflows/tests_lkcuda_cpp.yml:197]: Could you add a Note for this change? Feel free to add a url to the failed test for the future reference. thanks!

File: .github/CHANGELOG.md (hunk 1 above)
Comment (josephleekl) [reply .github/CHANGELOG.md:96]: Thanks Ali ... this bug has really destroyed me🤦

File: .github/workflows/tests_lkcuda_cpp.yml (hunk 2 above)
Comment (josephleekl) [reply .github/workflows/tests_lkcuda_cpp.yml:197]: I've added the link in the PR description, i'll also link it here https://github.com/PennyLaneAI/pennylane-lightning/blob/e9a3b80823a2dc52f747ff20f91031512d0dbd63/pennylane_lightning/core/src/simulators/lightning_kokkos/gates/tests/Test_StateVectorKokkos_Param.cpp#L2062

**Pull request merged**